src/display/annotation_layer.js

```diff
@@ -724,18 +728,36 @@ class PopupElement {
       popup.style.backgroundColor = Util.makeCssRgb(r | 0, g | 0, b | 0);
     }
 
-    let contents = this._formatContents(this.contents);
     let title = document.createElement('h1');
     title.textContent = this.title;
+    popup.appendChild(title);
+
+    // The modification date is shown in the popup instead of the creation
+    // date if it is available and can be parsed correctly, which is
+    // consistent with other viewers such as Adobe Acrobat. The localization
+    // string is equal to the one used for the document properties dialog
+    // for consistency.
+    try {
+      const dateObject = PDFDateString.toDateObject(this.modificationDate);
+      const modificationDate = document.createElement('span');
+      modificationDate.textContent = '{{date}}, {{time}}';
+      modificationDate.dataset.l10nId = 'document_properties_date_string';
```

Comment:
Something I forgot to mention previously: It's probably best to add a *new* string, rather than overloading an existing one in way.

Reply:
Yes, that's probably best. The commit is updated.

Reply:
Sorry, I didn't mean remove the existing one, but rather adding an *additional* new one for this use-case since locales may want different semantics depending on context.